File: Bubble_Sort/Bubble_Sort.php
<?php

function swap(&$arr,$i,$j){

	$temp = $arr[$i];
	$arr[$i] = $arr[$j];
	$arr[$j] = $temp;

	return;
}

function bubbleSort(&$arr){
	$isSorted = false;
	$lastUnsorted = count($arr)-1;
	while(!$isSorted) {
		$isSorted = true;
		for ($i=0; $i < $lastUnsorted; $i++) { 
			if ($arr[$i] > $arr[$i+1]) {
				swap($arr,$i,$i+1);
				$isSorted = false;
			}
		}
		$lastUnsorted--;
	}
	return;
}

function printArray($arr){
	$output = "[";
	for ($i=0; $i < count($arr); $i++) { 
		$output .= " ".$arr[$i]." ";
		if ($i<count($arr)-1) {
			$output .= ",";
		}
	}
	$output .= "]";

	echo $output;
	return;
}

$n = (int)readline();
$arr = array();

for ($i=0; $i < $n; $i++) { 
	$arr[$i] = (int)readline();
}

bubbleSort($arr);
printArray($arr);

?>

File: Bubble_Sort/Bubble_Sort.R
n = as.integer(readline())
arr = c()

for (i in 1:n) {
	arr[i] = as.integer(readline())
}


bubbleSort<-function(arr){
	isSorted = FALSE
	lastUnsorted = length(arr)-1
	while(!isSorted){
		isSorted = TRUE
		if(lastUnsorted>0){
		for (i in 1:lastUnsorted) {
		  cat(arr[i],arr[i+1],i,i+1,"\n")
			if(arr[i] > arr[i+1]){
				arr[c(i,i+1)] = arr[c(i+1,i)]
				isSorted = FALSE
			}
		}
		lastUnsorted = lastUnsorted - 1
		}
	}

	return(arr)

}

printArray<-function(arr){
	output = "["
	for (i in 1:length(arr)) {
		output = paste(output," ",arr[i]," ")
		if(i<length(arr))
			output = paste(output,",")
	}
	output = paste(output,"]")

	print(output)
}

arr = bubbleSort(arr)
printArray(arr)
